File: src/main/rst/examples/Chapter4/ruby/RemoteScreenShot.rb
require 'rubygems'
require 'selenium-webdrier'

begin
  driver = Selenium::WebDriver.for :remote, :url => "http://localhost:4444/wd/hub", :desired_capabilities => :firefox
  driver.get "http://www.google.com"
  driver.save_screenshot "/Screenshots/google.png"
ensure
  driver.quit
end

File: src/main/rst/examples/Chapter4/Javascript/RemoteScreenshot.js
var webdriver = require('selenium-webdriver');
var fs = require('fs');
var serverURL = 'http://localhost:4444/wd/hub';
var caps = {
    browserName: 'firefox'
}
var driver;
try {
    driver = new webdriver.Builder().usingServer(serverURL).withCapabilities(caps).build();
    driver.get('http://www.google.com');
    driver.takeScreenshot().then(function(data) {
        fs.writeFileSync('./google.png', new Buffer(data, 'base64'));
    })
} catch (e) {
    console.log('Exception occurred');
} finally {
    driver.quit();
}